src/models/geograph.py

```diff
@@ -518,22 +520,24 @@ def add_habitat(
                 invalid_indexes.append(node)
                 continue
 
-            polygon = polygons[node]
+            polygon = prep(polygons[node])
```

Comment:
I think we can move this into the else branch of your if-else below (line525) directly right? (not that it makes a big difference ^^)